updatersync/src/main/java/nl/openweb/hippo/groovy/GroovyFilesServiceImpl.java

```diff
@@ -99,25 +100,34 @@ private static boolean setUpdateScriptJcrNode(Node parent, File file) throws Rep
         final Updater updater = scriptClass.getUpdater();
         String name = updater.name();
         if(parent.hasNode(name)){
-            info("Updating existing script {}", name);
+            info("Updating existing script %s", name);
             parent.getNode(name).remove();
         }
         Node scriptNode = parent.addNode(name, HIPPOSYS_UPDATERINFO);
-
-        scriptNode.setProperty(HIPPOSYS_BATCHSIZE, updater.batchSize());
-        scriptNode.setProperty(HIPPOSYS_DESCRIPTION, updater.description());
-        scriptNode.setProperty(HIPPOSYS_PARAMETERS, updater.parameters());
-        scriptNode.setProperty(updater.xpath().isEmpty() ? HIPPOSYS_PATH : HIPPOSYS_QUERY,
-                updater.xpath().isEmpty() ? updater.path() : updater.xpath());
-        scriptNode.setProperty(HIPPOSYS_SCRIPT, scriptClass.getContent());
-        scriptNode.setProperty(HIPPOSYS_THROTTLE, updater.throttle());
+        final Map<String, Object> properties = PropertyCollector.getPropertiesForUpdater(scriptClass, file.getParentFile());
+        for (Map.Entry<String, Object> entry : properties.entrySet()) {
+            if(!JCR_PRIMARY_TYPE.equals(entry.getKey())){
+                scriptNode.setProperty(entry.getKey(), getValue(entry));
+            }
+        }
         return true;
     }
 
+    private static Value getValue(final Map.Entry<String, Object> entry) throws ValueFormatException {
```

Comment:
This won't ever be a multi-value field?

Reply:
All updater fields seem to be single value. So for now: no.